torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -344,29 +344,55 @@ void RequestCallbackImpl::processRpc(
         return;
       }
 
+      std::function<void(const c10::intrusive_ptr<c10::ivalue::Future>&)>
+          scriptPostProcessing;
+      if (scriptRemoteCall.isAsyncExecution()) {
+        scriptPostProcessing = [ownerRRef, postProcessing](
+            const c10::intrusive_ptr<c10::ivalue::Future>& jitFuture) {
+          try {
+            auto valueJitFuture = jitFuture->value().toFuture();
+            valueJitFuture->addCallback(
+                [valueJitFuture, ownerRRef, postProcessing]() {
+                  try {
+                    ownerRRef->setValue(valueJitFuture->value());
+                  } catch (const std::exception& e) {
+                    ownerRRef->setError(e.what());
+                  }
+                  postProcessing();
+                });
+          } catch (const std::exception& e) {
```

Comment:
Curious, what errors can happen here? Is it when the user async function does not return a future or is not properly decorated?